src/election.ts

```diff
@@ -0,0 +1,282 @@
+import BigNumber from 'bignumber.js';
+import * as EventEmitter from 'events';
+import { EtcdNoLeaderError, EtcdNotLeaderError } from './errors';
+import { Lease } from './lease';
+import { Namespace } from './namespace';
+
+/**
+ * Implmentation of etcd election.
+ * @see https://github.com/coreos/etcd/blob/master/clientv3/concurrency/election.go
+ *
+ * @example
+ * const client = new Etcd3()
+ * const election = new Election(client, 'singleton_service')
+ * const id = BigNumber.random().toString()
+ *
+ * // process will hang here until elected
+ * await election.campaign(id)
+ */
+export class Election extends EventEmitter {
+  public static readonly prefix = 'election';
+
+  private readonly namespace: Namespace;
+  private lease: Lease;
+
+  private leaseId = '';
+  private _leaderKey = '';
+  private _leaderRevision = '';
+  private _isCampaigning = false;
+  private _isObserving = false;
+
+  public get leaderKey(): string { return this._leaderKey; }
```

Comment:
cc @connor4312 Do we use `getters` anywhere in this project? This doesn't really follow the rest of the project's style.